script/install.sh

```diff
@@ -62,26 +62,43 @@ retry() {
         fi
     done
 }
-
 # url, path, [gpg signature url to verify]
+# url and the gpg signature can be an array of urls
 download() {
-    local url path
-    url=$1
-    path=$2
-
-    retry curl "$url" -o "$path"
+    local mirrors path
+    mirrors=($1)
+    path="$2"
+
+    try_all_mirrors() {
+        for i in "${!mirrors[@]}" ; do
+            if [ "$i" -gt 0 ] ; then
+                log "Falling back to mirror: ${mirrors[$i]}"
+            fi
+            if curl "${mirrors[$i]}" -o "$path" ; then
+                break
+            fi
+        done
+    }
+    retry try_all_mirrors
     if [ ! -z "${3:-}" ]; then
         verify "$3" "$path"
     fi
 }
 
-# url
+# url (can be an array of urls)
 fetch() {
-    local url path
-    url=$1
+    local mirrors path
+    mirrors=($1)
     path=$(mktemp "$TMP_ROOT/XXXXXX")
 
-    retry curl2 -sS "$url" -o "$path"
+    try_all_mirrors() {
+        for mirror in "${mirrors[@]}" ; do
+            if curl2 -sS "$mirror" -o "$path" ; then
```

Comment:
This is copy due to `curl2` being used here and `curl` in download. Is there any reason not to use `curl2` in  `download`?